recipes/lora_finetune_single_device.py

```diff
@@ -579,6 +584,9 @@ def train(self) -> None:
                     ):
                         break
 
+                    if curr_epoch == 0 and self.profiler_profile_memory and idx == self.profiler_wait_steps + self.profiler_warmup_steps:
+                        torch.cuda.memory._record_memory_history()
```

Comment:
The issue is if in the future we support other backend, like intel. It makes me think that this should be a function, instead of being hardcoded as "torch.cuda"

cc: @ebsmothers

Reply:
> The issue is if in the future we support other backend, like intel. It makes me think that this should be a function, instead of being hardcoded as "torch.cuda"
> 
> cc: @ebsmothers

I think it's OK, idk that memory snapshot would be supported on XPU anyways. We can do a check on device + profile_memory combo just to be safe though